### ISSUE TITLE
Add operators to cloud watch logs receiver

### ISSUE THREAD

**Issue by AndriySidliarskiy** [open-telemetry/opentelemetry-collector-contrib#20703]
### Is your feature request related to a problem? Please describe.

I have a body message in json format and need to parse it but I cannot do this with transform or filter operators massagemassageJSON

### Describe the solution you'd like

so maybe need to add operators into the cloud watch receiver

**Comment by AndriySidliarskiy** [comment]
@astencel-sumo thanks for the answer. I have a json body message 
` . 
logRecord.body:{"id":"218","type":"TaskStarted","details":{"resource":"call","resourceType":"eks"},"previous_event_id":"217","event_timestamp":"1680697114137","execution_arn":"arn:aws:states:eu-west-1:888888888:execution:PVCStateMachine-fFgbdDPOdoMv:47ec3aca-e90f-4d50-8981-fb60a655cce1"}`
that is going from cloud watch receiver and I need to parse this message if it`s possible. I ty to use filter with this setting ```
 # filter/cw:
    #   error_mode: ignore
    #   logs:
    #     log_record:
    #       - ParseJSON(body) 
```
but I have error `2023/04/06 09:54:04 collector server run finished with error: invalid configuration: processors::filter/cw: unable to parse OTTL statement: 1:29: unexpected token "<EOF>" (expected <opcomparison> Value) 

And In the transform processor not possible to use ParseJSON function or I cannot see an example.

**Comment by AndriySidliarskiy** [comment]
it looks like what i need but it in development https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/logstransformprocessor

**Comment by astencel-sumo** [comment]
Here's the configuration for the [Transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.75.0/processor/transformprocessor) and the ParseJSON function that works for me with the latest version of the collector (v0.75.0):

```yaml
exporters:
  logging:
    verbosity: detailed

processors:
  transform/parse-json:
    log_statements:
    - context: log
      statements:
      - set(body, ParseJSON(body))

receivers:
  filelog:
    include: ./logs-json.log
    start_at: beginning

service:
  pipelines:
    logs:
      exporters:
      - logging
      processors:
      - transform/parse-json
      receivers:
      - filelog
```

You can also use `set(attributes, ParseJSON(body))` to put the parsed data in the record attributes. Please let me know if this points you in the right direction and whether you have any further questions.

I admit that the documentation of the Transform processor could use some improvements. Pull requests are very welcome! 😅 

The Logs Transform processor is intended to be made obsolete by the Transform processor (see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/19775), so it will probably never go out of the "development" stage, other than into "deprecated".

**Comment by AndriySidliarskiy** [comment]
@astencel-sumo thanks for answer but again error.```
 Error: invalid configuration: processors::transform/cw: error while parsing arguments for call to 'set': invalid argument at position 0: invalid resource path expression [{body <nil>}]
2023/04/06 14:45:09 collector server run finished with error: invalid configuration: processors::transform/cw: error while parsing arguments for call to 'set': invalid argument at position 0: invalid resource path expression [{body <nil>}]
```

**Comment by astencel-sumo** [comment]
What is the version of the collector that you're using? Also please paste the configuration you're using.

**Comment by AndriySidliarskiy** [comment]
oh i see my problem, i use context resource instead log) now working) thanks a lot

**Comment by AndriySidliarskiy** [comment]
@astencel-sumo does it`s possible again to parse value but reproduce when convert to JSON body message?

`logRecord.body.details.output:{Pvc:{Size:1200Gi,Operation-type:create,Name:SubjectId},NeedCallback:true,AWS_STEP_FUNCTIONS_STARTED_BY_EXECUTION_ID:arn:aws:states:eu-west-1:8888888888:execution:Workload369lxpK:20230404-1422-Project1-Study1-Cohort1-LabId1-SubjectId-NormalSampleId1,Endpoint:https://23EA738A075020EC313CA85D4D97B553.yl4.eu-west-1.eks.amazonaws.com,Namespace:cs-ari-pvc2-euwe1},Namespace:cs-ari-pvc2-euwe1,PvcName:subjectid-normalsampleid132323232323,error:{Error:EKS.404,Cause:{\ResponseBody\:{\kind\:\Status\,\apiVersion\:\v1\,\metadata\:{},\status\:\Failure\,\message\:\workflows.argoproj.io \\\subjectid-normalsampleid132323232323\\\ not found\,\reason\:\NotFound\,\details\:{\name\:\subjectid-normalsampleid132323232323\,\group\:\argoproj.io\,\kind\:\workflows\},\code\:404},\StatusCode\:404,\StatusText\:\Not Found\}},pvc:{workflowStatus:{phase:Running}}}`

**Comment by astencel-sumo** [comment]
> parse value but reproduce when convert to JSON body message?

I'm sorry, I don't understand. Can you elaborate?

**Comment by AndriySidliarskiy** [comment]
previously i parse my body with ParjeJSON function ann all good but it creates another attribute with the name logRecord.body.details.output in JSON format and I must again convert this from json format. Does it`s possible?

**Comment by astencel-sumo** [comment]
Let's assume this is your input string:

`{"key1":"value1","details":{"output":{"pvc":{"size":"1200Gi","operation-type":"create"}}}}`

After applying the `ParseJSON` function, your output is a map:

```json
{
  "key1": "value1",
  "details": {
    "output": {
      "pvc": {
        "size": "1200Gi",
        "operation-type": "create"
      }
    }
  }
}
```

Is your expectation to keep the value of `output` as a string?

```json
{
  "key1": "value1",
  "details": {
    "output": "{\"pvc\":{\"size\":\"1200Gi\",\"operation-type\":\"create\"}}"
  }
}
```

**Comment by AndriySidliarskiy** [comment]
so i have a body in JSON format after I parsed this body it automatically creates a new attribute with a value logRecord.body.details.output that contains a value in JSON format, and I need to parse this again. For example 

{
  "key1": "value1",
  "details": {
    "output": {{\"pvc\":{\"size\":\"1200Gi\",\"operation-type\":\"create\"}} }
  }
}

**Comment by astencel-sumo** [comment]
OK yeah, so it looks like there's something after the JSON parser that expects the `body.details.output` field to contain a string and not a map.

Perhaps if possible, you could try to change that "something" that is throwing the "expected string but got pcommon.Map" error to expect a map instead?

Other that that, I don't see a solution for it with the Transform processor. I think it's best to open a new issue describing this specific problem - e.g. "[pkt/ottl] how to partially parse JSON". This will make it easier for others to take part in the discussion without digging through comments that are not relevant to the problem.

**Comment by AndriySidliarskiy** [comment]
if i understood this value inside pcommon. map so json cannot parse pcommon.map, he waits on a string.
